sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -847,21 +847,32 @@ class Analyzer(
       case _ => plan
     }
 
-    def apply(plan: LogicalPlan): LogicalPlan = ResolveTempViews(plan).resolveOperatorsUp {
-      case i @ InsertIntoStatement(table, _, _, _, _) if i.query.resolved =>
-        val relation = table match {
-          case u @ UnresolvedRelation(SessionCatalogAndIdentifier(catalog, ident)) =>
-            lookupRelation(catalog, ident, recurse = false).getOrElse(u)
-          case other => other
-        }
+    def apply(plan: LogicalPlan): LogicalPlan = {
+      var logicalPlans = Map.empty[UnresolvedRelation, LogicalPlan]
```

Comment:
Why don't you use a mutable map while we're here? Also, can you add a short comment with the JIRA ID?

Reply:
Add comment to doc:
https://github.com/apache/spark/blob/07e6b7f7529f07a73dae9baede5e2046a18e35bc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L94-L96